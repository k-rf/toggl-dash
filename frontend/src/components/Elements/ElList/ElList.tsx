import { List, ListProps } from "@mui/material";

type Props = Pick<ListProps, "children" | "sx">;

export const ElList = (props: Props) => {
  return <List {...props} />;
};
