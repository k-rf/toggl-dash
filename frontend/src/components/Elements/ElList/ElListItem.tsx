import { ListItem, ListItemProps } from "@mui/material";

type Props = Pick<ListItemProps, "children" | "sx">;

export const ElListItem = (props: Props) => {
  return <ListItem {...props} />;
};
