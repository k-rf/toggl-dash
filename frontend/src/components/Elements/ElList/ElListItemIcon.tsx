import { ListItemIcon, ListItemIconProps } from "@mui/material";

type Props = Pick<ListItemIconProps, "children">;

export const ElListItemIcon = (props: Props) => {
  return <ListItemIcon {...props} />;
};
