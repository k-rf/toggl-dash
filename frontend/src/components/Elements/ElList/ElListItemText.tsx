import { ListItemText, ListItemTextProps } from "@mui/material";

type Props = Pick<ListItemTextProps, "children">;

export const ElListItemText = (props: Props) => {
  return <ListItemText {...props} />;
};
