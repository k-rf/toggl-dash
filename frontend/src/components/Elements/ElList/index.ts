export * from "./ElList";
export * from "./ElListItem";
