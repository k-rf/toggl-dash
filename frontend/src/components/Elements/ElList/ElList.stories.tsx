import type { Meta, StoryObj } from "@storybook/react";

import { ElList } from "./ElList";
import { ElListItem } from "./ElListItem";

export default { component: ElList } as Meta<typeof ElList>;

type Story = StoryObj<typeof ElList>;

export const Default: Story = {
  args: {
    children: (
      <>
        <ElListItem>List Item A</ElListItem>
        <ElListItem>List Item B</ElListItem>
        <ElListItem>List Item C</ElListItem>
      </>
    ),
  },
};
