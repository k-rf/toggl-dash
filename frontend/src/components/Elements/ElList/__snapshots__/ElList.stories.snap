// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Elements/ElList/ElList > Default 1`] = `
<ul
  className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    List Item A
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    List Item B
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
  >
    List Item C
  </li>
</ul>
`;
